app/Http/Controllers/PostController.php

```diff
@@ -2,13 +2,168 @@
 
 namespace App\Http\Controllers;
 
-use App\Models\Post;
+use App\Contracts\Services\PostServiceInterface;
 use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
 
 class PostController extends Controller
 {
-    public function index()
+    private $postService;
+
+    public function __construct(PostServiceInterface $postService)
+    {
+        $this->postService = $postService;
+    }
+    
+    /**
+     * create function
+     *
+     * @return post create view
+     */
+    public function create()
+    {
+        return view('posts.create'); 
+    }
+
+    /**
+     * store function
+     *
+     * @return post list view
+     */
+    public function store(Request $request)
+    {
+        $validated = $request->validate([
+            'title'       => 'required|string|max:255',
+            'description' => 'nullable|string',
+            'public_flag' => 'required|boolean',
+        ]);
+
+        $validated['user_id'] = auth::id();
+
+        // Ensure public_flag is 1 or 0 (from radio button)
+        $validated['public_flag'] = $request->input('public_flag') ? 1 : 0;
+
+        // Use service to store
+        $this->postService->storePost($validated);
+        // Redirect based on role
+        $currentUser = Auth::user();
+        if ($currentUser->role == 1) {
+            return redirect()->route('users.list')->with('status', 'Post created successfully!');
+        } else {
+            return redirect()->route('users.show', $currentUser->id)->with('status', 'Post created successfully!');
+        } 
+    }
+
+    /**
+     * postList function
+     *
+     * @return post list view
+     */
+    public function postList(){
+        $posts = $this->postService->getAllPosts();
+        return view('posts.list', compact('posts'));        
+    }
+    
+    /**
+     * show function
+     *
+     * @param integer $id
+     * @return post details view
+     */
```

Comment:
```suggestion
    /**
     * show user's post detail
     *
     * @param integer $id
     * @return View
     */
```

make docs more specific and also fix other functions